web/app/profile/security/page.tsx

```diff
@@ -40,10 +42,14 @@ const defaultShowPassword = {
 };
 
 const SecurityPage = observer(() => {
+  // store
+  const { data: currentUser } = useUser();
   // states
   const [showPassword, setShowPassword] = useState(defaultShowPassword);
   const [isPasswordInputFocused, setIsPasswordInputFocused] = useState(false);
   const [isRetryPasswordInputFocused, setIsRetryPasswordInputFocused] = useState(false);
+  const [oldPasswordRequired, setOldPasswordRequired] = useState(!currentUser?.is_password_autoset);
```

Comment:
Why do we need to use a state over here? Let's just directly update `currentUser?.is_password_autoset` once the password is added?